index.js

```diff
@@ -381,5 +381,32 @@ form.addEventListener('submit', (event) => {
     error.textContent = '';
     error.className = 'error';
     form.submit();
+    localStorage.removeItem('data');
   }
-});
\ No newline at end of file
+});
+
+// Preserve data in web browser
+
+// Adding data to local storage
+form.addEventListener('keyup', () => {
+  const data = {
+    name: document.getElementById('full-name').value,
+    email: document.getElementById('email').value,
+    message: document.getElementById('message').value,
+  };
+  localStorage.setItem('data', JSON.stringify(data));
+});
+
+function fillForm() {
+  const data = JSON.parse(localStorage.getItem('data'));
+  if (data) {
+    const name = form.elements['full name'];
+    const { email } = form.elements;
+    const { message } = form.elements;
+    name.value = data.name;
+    email.value = data.email;
+    message.value = data.message;
+  }
+}
```

Comment:
- [ ] As required for the project, kindly make sure your contact form follows the flowing aspects:
  - Data can be used after being retrieved from the local storage the saved string is turned back into an object for that you can use `JSON.parse`
  - When the user loads the page, the code checks if there are any saved data in the local storage and if so the input fields are pre-filled with the data. e.g.:
   ```JavaScript
      window.onload = () => {
         // Your code here
       };
   ```